staging/src/k8s.io/apimachinery/pkg/runtime/error.go

```diff
@@ -120,3 +120,77 @@ func IsMissingVersion(err error) bool {
 	_, ok := err.(*missingVersionErr)
 	return ok
 }
+
+// StrictDecoderError is a base error type that is returned by a strict Decoder such
+// as UniversalStrictDecoder.
+type StrictDecoderError struct {
+	message      string
+	gvk          schema.GroupVersionKind
+	originalData []byte
+}
+
+// UnknownFieldError is an error type that is returned in cases where the input
+// JSON or YAML contains unknown fields.
+// UnknownFieldError embeds StrictDecoderError.
+type UnknownFieldError struct {
```

Comment:
if we don't have good library support to detect this or the duplicate field case, let's not expose these typed errors yet

Reply:
+1 on keeping only the StrictDecoderError, as per my original proposal.